### ISSUE TITLE
Elasticsearch 7 support

### ISSUE THREAD

**Issue by laevilgenius** [sabricot/django-elasticsearch-dsl#170]
This depends on #115, although a few more changes are required, the most notable being the [removal](https://www.elastic.co/guide/en/elasticsearch/reference/7.0/removal-of-types.html) of mapping types.
To make things easier, perhaps it makes sense to mimic `elasticsearch-dsl` versioning scheme?

**Comment by stylemistake** [comment]
Just adding my two cents, this guy already has 7.0 working in his own repo (with few API changes along the way).

https://github.com/laevilgenius/django-elasticsearch-dsl

Although it is still a work in progress, it works like a charm. If author is not going to provide any updates, I suggest y'all consider this fork as a new upstream.

**Comment by laevilgenius** [comment]
@stylemistake, thanks, but I don't deserve the credit here. Most of the work was done in #136, which my fork is based on. I just removed the mapping types and added a few utility functions.

**Comment by safwanrahman** [comment]
> Although it is still a work in progress, it works like a charm. If author is not going to provide any updates, I suggest y'all consider this fork as a new upstream.
> 
> P.S. Good job, mate.

As one of the contributor of this project and sole contributor of #136, which this fork is based on, I think your comment is disrespectful to the maintainer of this project. You should show empathy towards the maintainers who are giving their time for this project.

**Comment by stylemistake** [comment]
I am sorry for commenting in this poor form. All I wanted to convey is that his branch contained all relevant PRs which make it all work.

**Comment by safwanrahman** [comment]
Thanks for the issue @laevilgenius . I have started working on this in #176. Can you suggest some changes that you made in your fork?

**Comment by laevilgenius** [comment]
@safwanrahman, here's the [diff](https://github.com/laevilgenius/django-elasticsearch-dsl/commit/118e9bc824e4b23bc712316105a5db7640674add). It's pretty straightforward: I removed mapping types and made code compatible with elasticsearch-dsl>=7.0.0. I also fixed an issue with tests not deleting test indices, although I'm still not sure it works every time.

Plus I added a couple of utility functions which I find quite useful, but I have already submitted them as PRs.
